modules/nf-core/untar/main.nf

```diff
@@ -11,16 +11,19 @@ process UNTAR {
     tuple val(meta), path(archive)
 
     output:
-    tuple val(meta), path("$prefix"), emit: untar
-    path "versions.yml"             , emit: versions
+    tuple val(meta), path("$prefix")            , emit: untar
+    tuple val(meta), path("${prefix}_files/**") , emit: files, optional: true
+    path "versions.yml"                         , emit: versions
 
     when:
     task.ext.when == null || task.ext.when
 
     script:
     def args  = task.ext.args ?: ''
     def args2 = task.ext.args2 ?: ''
+    def args3 = task.ext.args3 ?: ''
```

Comment:
I think this could be a boolean input. This seems a bit of weird workaround in my opinion. But again this is a difficult situation, we should get some more maintainer and core team eyes on it 😁

Reply:
Thanks! Could you clarify on how you think that boolean would be supplied?

Reply:
As an actual input value. That seems the best solution here, but it's a bit of a grey line for the guidelines 😁

Reply:
Well, it's non-mandatory as currently implemented (default is not to emit the files, as per current behaviour). So I don't think it qualifies for the `val` treatment, right?

Reply:
Yeah but it's not an argument for a tool either... (oh god this will probably start another discussion 😁)

Reply:
I could just revert the conditionality and emit the files all the time....

Reply:
That's also a possibility, would you know if nextflow resolves the output channels when the module finishes or when they are actually used?

Reply:
I don't really see how 'lazy globbing' might work, so I think we have to assume files channel would get filled from the glob regardless.

Reply:
Yeah true better safe than sorry

Reply:
Blah, I really don't understand the issue with adding other options that aren't the default set. They just make the code more readable.

Reply:
I agree especially in this case!